### PR TITLE
have the shows scripts keybinding work better across platforms

### PR DESCRIPTION
**Pull request by devoncarew**
have the shows scripts keybinding work better across platforms:
- implement platform detection for desktop and flutter web
- add the ability to render a keybinding to user-facing text
- prevent the browser's default behavior on the mac for the command-P keybinding

We now show `⌘P` in the `Libraries` filter text field; some work towards https://github.com/flutter/devtools/issues/1972.

### CODE REVIEW

File: packages/devtools_app/lib/src/config_specific/host_platform/host_platform_stub.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file.
+
+// ignore: avoid_classes_with_only_static_members
+class HostPlatform {
+  // ignore: prefer_const_declarations
+  static final HostPlatform instance = null;
+
+  bool get isMacOS => false;
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply packages/devtools_app/lib/src/config_specific/host_platform/host_platform_stub.dart:10]: 👍 will do

**Comment by devoncarew** [comment]
See https://github.com/flutter/flutter/issues/58119 for the issue tracking preventing the browser default keybindings.

File: packages/devtools_app/lib/src/config_specific/flutter/framework_initialize/_framework_initialize_web.dart [hunk 2]
```diff
@@ -25,6 +25,23 @@ Future<String> initializePlatform() async {
...
 
+  // Prevent the browser default behavior for specific keybindings we'll later
+  // handle in the app. This is a workaround for
+  // https://github.com/flutter/flutter/issues/58119.
+  window.onKeyDown.listen((event) {
+    if (!event.metaKey) {
+      return;
+    }
+
+    if (!window.navigator.userAgent.contains('Macintosh')) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/config_specific/flutter/framework_initialize/_framework_initialize_web.dart:36]: maybe comment on why the default is only prevented on mac.

Comment (devoncarew) [reply packages/devtools_app/lib/src/config_specific/flutter/framework_initialize/_framework_initialize_web.dart:36]: Will do -

**Pull request merged**